### PR TITLE
use Task.Run to execute SmtpClient

### PR DESCRIPTION
**Pull request by wfurt**
I'm not sure why #74545 did not work. we got one more test hanging instance after it. 
After looking at the Mail code the `SendMailAsync` actually does fair amount of synchronous processing before returning the `Task`. 
I'm not sure if there is better way to do it @stephentoub but I'm pushing that to the thread-pool to be sure the code before it and final `await` can execute.

This is only one test in `Mail` test suite using the `ManualResetEvent`. I did look at and it seems ok to me. We basically block completion so the client has chance to cancel. 
 
contributes to https://github.com/dotnet/runtime/issues/73447

### CODE REVIEW

File: src/libraries/System.Net.Mail/tests/Functional/SmtpClientTest.cs [hunk 1]
```diff
@@ -473,7 +473,7 @@ public async Task SendMailAsync_CanBeCanceled_CancellationToken()
 
             var message = new MailMessage("foo@internet.com", "bar@internet.com", "Foo", "Bar");
 
-            Task sendTask = client.SendMailAsync(message, cts.Token);
+            Task sendTask = Task.Run(() => client.SendMailAsync(message, cts.Token));
```
[the remark this answers is not in the document]
Comment (wfurt) [reply src/libraries/System.Net.Mail/tests/Functional/SmtpClientTest.cs:476]: The whole test hangs and I have no visibility why. When this happens Helix just kills the suite and we get no information.  In https://github.com/dotnet/runtime/pull/74545 I tried to add cap on how long we `await` for tasks - but that does not seems to be sufficient. So it feels we get stuck somewhere while creating the `sendTask `. So the goal is to put this to background so we can get to the `await` bellow and timeout ... and possibly collect some information. this is _not_  attempt to fix the root cause.

File: src/libraries/System.Net.Mail/tests/Functional/SmtpClientTest.cs [hunk 2]
```diff
@@ -482,7 +482,7 @@ public async Task SendMailAsync_CanBeCanceled_CancellationToken()
             await Assert.ThrowsAsync<TaskCanceledException>(async () => await sendTask).WaitAsync(TestHelper.PassingTestTimeout);
 
             // We should still be able to send mail on the SmtpClient instance
-            await client.SendMailAsync(message).WaitAsync(TestHelper.PassingTestTimeout);
+            await Task.Run(() => client.SendMailAsync(message).GetAwaiter().GetResult()).WaitAsync(TestHelper.PassingTestTimeout);
```
Comment (MihaZupan) [inline src/libraries/System.Net.Mail/tests/Functional/SmtpClientTest.cs:485]:
```suggestion
            await Task.Run(() => client.SendMailAsync(message)).WaitAsync(TestHelper.PassingTestTimeout);
```

File: src/libraries/System.Net.Mail/tests/Functional/SmtpClientTest.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (stephentoub) [reply src/libraries/System.Net.Mail/tests/Functional/SmtpClientTest.cs:476]: > this is not attempt to fix the root cause.

Got it, ok, thanks.

**Review by rzikm — APPROVED** [review]
LGTM

**Pull request merged**